.github/workflows/build-windows.yml

```diff
@@ -4,75 +4,308 @@ on:
   push:
     branches-ignore:
       - 'coverityScan'
+  pull_request:
+    branches:
+      - 'main'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.job }}-${{ github.ref }}
+  cancel-in-progress: true
 
 jobs:
   build-windows:
-    runs-on: ${{matrix.os}}
+    name: '${{ matrix.os }} (msvc ${{ matrix.arch }}, ${{ matrix.cpp_std }})'
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [windows-2019]
-        arch: ['amd64', 'x86']
-        coverage: [false, true]
+        os:
+          - windows-2019
+          - windows-2022
+        arch: 
+          - x86
+          - amd64
+        cpp_std:
+          - 'c++14'
+          - 'c++17'
       fail-fast: false
     env:
       CC: cl.exe
       CXX: cl.exe
       LD: link.exe
     steps:
-      - name: Chekcout crunch
-        uses: actions/checkout@v2
+      - name: Runtime environment
+        shell: bash
+        env:
+          WORKSPACE: ${{ github.workspace }}
+        run: |
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+          echo "GITHUB_WORKSPACE=`pwd`" >> $GITHUB_ENV
+      - name: Setup compiler
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: ${{ matrix.arch }}
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          lfs: true
+          submodules: true
+      - name: Setup Meson + Ninja
+        shell: bash
+        run: |
+          python3 -m pip install --upgrade pip setuptools wheel
+          python3 -m pip install meson ninja
+        working-directory: ${{ runner.temp }}
+      - name: Version tools
+        shell: bash
+        run: |
+          $CC  || true
+          $LD  || true
+          meson --version
+          ninja --version
+      - name: Configure
+        run: meson build --prefix=$HOME/.local -Dcpp_std=${{ matrix.cpp_std }} --buildtype=debug
```

Comment:
Is passing `--buildtype=debug` here vestigial or intentional?

Reply:
Probably intentional, as we spoke on private, but I can't recall the specific reason why we used it as such in Substrate. Removing.